### PR TITLE
Pinvoke testing

### PR DESCRIPTION
**Pull request by KevinRansom**
Enable Pinvoke testing on coreclr.

Because pinvoke on coreclr is not supported on versions of dotnet framework lower than 3.0, we generate a flag at build time that determines whether to run the test or not.

Also adds a flag generated at build time, that lets us know whether we are running on coreclr version 3.0.
The tests run on desktop and coreclr, when applicable.

Some code was removed from the original test cases, this code was removed because it depends on a windows dll, cards.dll that is no longer shipped.  And other code that relied on popdyn.dll, that I have no idea what it is or where it is.  The existing test merely compiled against them.  Now we also run the tests, so at last they are actually testing pinvoke.

### CODE REVIEW

**Comment by NinoFloris** [comment]
This sounds too simple? I've used p/invoke since dotnet core 1.0 without problems. I'm not aware of big changes there either for 3.0?

**Comment by cartermp** [comment]
> dotnet framework lower than 3.0

I'm confused. So it's compatible with .NET Framework?

**Comment by KevinRansom** [comment]
@cartermp, @NinoFloris this is pinvoke from **fsi**, it worked fine for compiled apps using fsc.exe.  The RefEmit API for emitting the necessary code to support pinvoke wasn't added until netcoreapp3.0.

The desktop FSI has always supported pinvoke from **fsi**.  The coreclr version of **fsi** has not supported pinvoke until this PR:  https://github.com/Microsoft/visualfsharp/pull/6542

I hope this clears things up

Kevin

File: src/fsharp/FSComp.txt [hunk 1]
```diff
@@ -1458,3 +1458,4 @@ notAFunctionButMaybeDeclaration,"This value is not a function and cannot be appl
 3245,tcCopyAndUpdateNeedsRecordType,"The input to a copy-and-update expression that creates an anonymous record must be either an anonymous record or a record"
 3300,chkInvalidFunctionParameterType,"The parameter '%s' has an invalid type '%s'. This is not permitted by the rules of Common IL."
 3301,chkInvalidFunctionReturnType,"The function or method has an invalid return type '%s'. This is not permitted by the rules of Common IL."
+useSdkRefs,"Use reference assemblies for .NET framework references when available (Enabled by default))."
```
Comment (brettfo) [inline src/fsharp/FSComp.txt]: Mismatched parentheses at the end.

Comment (brettfo) [reply src/fsharp/FSComp.txt]: Will also require updating xlf files.

File: tests/FSharp.Compiler.UnitTests/VersionCompare.fs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    [<TestCase("1.0.1", "1.0.0", ExpectedResult = 1)>]                  // 1.0.1 > 1.0.1
+    [<TestCase("0.0.9", "1.0.0-Suffix1", ExpectedResult = -1)>]         // 0.0.9 < 1.0.0-Suffix1
+    [<TestCase("1.0.0", "1.0.0-Suffix1", ExpectedResult = 1)>]          // 1.0.0 > 1.0.0-Suffix1
+    [<TestCase("1.0.0-Suffix1", "1.0.0", ExpectedResult = -1)>]         // 1.0.0-Suffix1 < 1.0.0
+    [<TestCase("1.0.0-Suffix1", "1.0.0-Suffix2", ExpectedResult = -1)>] // 1.0.0-Suffix1 < 1.0.0-Suffix2
+    [<TestCase("1.0.0-Suffix2", "1.0.0-Suffix1", ExpectedResult = 1)>]  // 1.0.0-Suffix2 > 1.0.0-Suffix1
+    [<TestCase("1.0.0-Suffix1", "1.0.0-Suffix1", ExpectedResult = 0)>]  // 1.0.0-Suffix1 > 1.0.0-Suffix2
+    [<TestCase("1.0.1", "1.0.0-Suffix1", ExpectedResult = 1)>]          // 1.0.1 > 1.0.0-Suffix1
+    [<TestCase("1.0.0-Suffix1", "1.0.1", ExpectedResult = -1)>]        // 1.0.0-Suffix1 < 1.0.1
+    let VersionCompareTest (str1 : string, str2 : string) : int =
```
Comment (brettfo) [inline tests/FSharp.Compiler.UnitTests/VersionCompare.fs]: Whitspace.  E.g.,

``` diff
-let VersionCompareTest (str1 : string, str2 : string) : int =
+let VersionCompareTest (str1: string, str2: string): int =
```

Other instances in this file, too.

**Comment by cartermp** [comment]
@KevinRansom Did you intend to merge this? Build shows failure: https://dev.azure.com/dnceng/public/_build/results?buildId=165027

**Comment by KevinRansom** [comment]
@cartermp bugger, it has some extra and buggy code in it.

**Pull request merged**